cmd/snap-seccomp/main_test.go

```diff
@@ -350,8 +351,20 @@ func (s *snapSeccompSuite) TestUnrestricted(c *C) {
 //    {"read >=2", "read;native;1", main.SeccompRetKill},
 //    {"read >=2", "read;native;0", main.SeccompRetKill},
 func (s *snapSeccompSuite) TestCompile(c *C) {
-	// The 'shadow' group is different in different distributions
-	shadowGid, err := osutil.FindGid("shadow")
+	// The 'shadow' group is different in different distributions, if group
+	// does not exist, instead of guessing gid based on group name, look at
+	// /etc/shadow and use the gid that owns it
+	var shadowGid uint64
+	if gid, err := osutil.FindGid("shadow"); err != nil {
+		var stat syscall.Stat_t
+		err := syscall.Stat("/etc/shadow", &stat)
```

Comment:
I think we could twist this test a little by just stating `/etc/shadow` unconditionally and using whatever the group there for the rest of the code below. Alternatively use something that works on Arch as well and just carry on.

The real problem is in the `account_control` interface which grants fchown to `u:root` and `g:shadow`. This will fail at runtime and needs to be fixed to grant the actual group owner of `/etc/shadow`

Reply:
Agreed, skipping the fallback and doing `stat()` unconditionally will make the code easier. I'll amend the PR.